source/bson/documents.txt

```diff
@@ -0,0 +1,242 @@
+     - ``Double``
+
+   * - ``Int32``
+     - ``Integer``
+
+   * - ``Int64``
+     - ``Long``
+
+   * - ``String``
+     - ``String``
+
+   * - ``Binary``
+     - ``Array[Byte]``
+
+   * - ``ObjectId``
+     - ``ObjectId``
+
+   * - ``Null``
+     - ``None``
+
+It is possible to change or extend these mappings, a process described
+in the following sections.
+
+The following sections describe the two main ``Document`` classes.
+
+Immutable Documents
+-------------------
+
+Similar to the Scala collections library, the immutable class is the preferred
+class. For convenience, it is aliased to ``org.mongodb.scala.Document`` and
+``org.mongodb.scala.bson.Document`` as well as being available from
+``org.mongodb.scala.bson.collection.immutable.Document``.
+
+Instances of this
+type are guaranteed to be immutable for everyone. Such a collection will
+never change after it is created. Therefore, you can rely on the fact
+that accessing the same collection value repeatedly at different points
+in time will always yield a collection with the same elements.
+
+.. code-block:: scala
+
+   import org.mongodb.scala.bson._
+   
+   val doc1 = Document("AL" -> BsonString("Alabama"))
+   val doc2 = doc1 + ("AK" -> BsonString("Alaska"))
+   val doc3 = doc2 ++ Document("AR" -> BsonString("Arkansas"), "AZ" -> BsonString("Arizona"))
+
+Mutable Documents
+-----------------
+
+To get the mutable ``Document`` type, you need to import it explicitly
+from ``org.mongodb.scala.collections.mutable.Document``. The mutable
+``Document`` can be updated or extended in place. This means you can change,
+add, or remove elements of the ``Document`` as a side effect. Similar to Scala
+collections, when dealing with mutable types you need to understand
+which code changes which collection and when.
+
+.. code-block:: scala
+
+   import org.mongodb.scala.bson._
+   import org.mongodb.scala.bson.collection.mutable.Document
+   
+   val doc = Document("AL" -> BsonString("Alabama"))
+   val doc1 = doc + ("AK" -> BsonString("Alaska"))   // doc not mutated but new doc created
+   doc1 ++= Document("AR" -> BsonString("Arkansas"), 
+                     "AZ" -> BsonString("Arizona"))  // doc1 mutated as ++= changes in place.
+
+Implicit Conversions
+--------------------
+
+For many of the ``BsonValue`` types, there are obvious direct mappings from a
+Scala type. For example, a ``String`` maps to ``BsonString``, an ``Int`` maps to
+``BsonInt32`` and a ``Long`` maps to a ``BsonInt64``. For convenience, these types
+can be used directly with ``Document`` types and they are converted by the
+contract traits in the ``BsonMagnets`` object. As long as there is an
+implicit ``BsonTransformer`` in scope for any given type, then that type can
+be converted into a ``BsonValue``.
+
+The following ``BsonTransformers`` are in scope by default:
+
+.. list-table::
+   :header-rows: 1
+   :class: compatibility-large
+
+   * - Scala Type
+     - BsonValue
+
+   * - ``Boolean``
+     - => ``BsonBoolean``
```

Comment:
Suggestion:

I think "=>" should be omitted. In the [original table](https://mongodb.github.io/mongo-java-driver/4.11/driver-scala/bson/scala-documents/), I think it is used to show the mapping, but is not be part of the "BsonValue" column.